File: apps/php-sdk/src/Models/AgentTraceActiveBrowserSession.php
<?php

declare(strict_types=1);

namespace Firecrawl\Models;

final class AgentTraceActiveBrowserSession
{
    public function __construct(
        private readonly ?string $id = null,
        private readonly ?string $liveViewUrl = null,
        private readonly ?AgentTraceViewport $viewport = null,
    ) {}

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'] ?? null,
            liveViewUrl: $data['liveViewUrl'] ?? null,
            viewport: isset($data['viewport']) && is_array($data['viewport'])
                ? AgentTraceViewport::fromArray($data['viewport'])
                : null,
        );
    }

    public function getId(): ?string
    {
        return $this->id;
    }

    public function getLiveViewUrl(): ?string
    {
        return $this->liveViewUrl;
    }

    public function getViewport(): ?AgentTraceViewport
    {
        return $this->viewport;
    }
}

File: apps/java-sdk/src/main/java/com/firecrawl/models/AgentTraceActiveBrowserSession.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * An active browser session attached to an agent run, with a live view URL.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AgentTraceActiveBrowserSession {

    private String id;
    private String liveViewUrl;
    private AgentTraceViewport viewport;

    public String getId() { return id; }
    public String getLiveViewUrl() { return liveViewUrl; }
    public AgentTraceViewport getViewport() { return viewport; }

    @Override
    public String toString() {
        return "AgentTraceActiveBrowserSession{id=" + id + "}";
    }
}
